parlai/agents/fairseq/fairseq.py

```diff
@@ -477,31 +499,32 @@ def report(self):
         return m
 
     def reset_metrics(self):
+        """Reset metrics calculated by the model back to zero."""
         if not hasattr(self, "trainer"):
-            # We haven't initialized the trainer yet, so we don't have any metrics
+            # We haven't set up the trainer yet, so we don't have any metrics
             return
         # We need to reset everything
         for k in self.trainer.meters:
             self.trainer.meters[k].reset()
 
     def receive_metrics(self, metrics_dict):
-        """Used to update lr scheduler."""
+        """Update lr scheduler with validation loss."""
         self.trainer.lr_step(-1, metrics_dict["valid_loss"])
 
     # Helper functions
     def _seq_length(self, xs):
-        """Computes length of the sequence (non-padded size)"""
+        """Compute length of the sequence (non-padded size)."""
```

Comment:
I think we can delete this whole method now that the seq length is coming in on the Batch. I'll wait to do this after this patch lands though.